### PR TITLE
Split Door Bolt functionality out of AirlockDoor

### PR DESCRIPTION
**Pull request by tom-leys**
## About the PR
- The code controlling bolting / unbolting has been split out of AirlockDoorComponent into DoorBoltComponent
- AirlockDoorSystem has also been split
Note: I've simplified an earlier PR #16189 : Here we have JUST the refactoring with no new functionality. Expect code moving and changing from `AirlockSystem` to `DoorBoltSystem`

**Media**

A little demo that some things have not changed:
- Wire cutting and multitooling deploys / toggles bolts
- Door remote works
- Bolts block crowbar (jaws of life)

https://github.com/space-wizards/space-station-14/assets/5285589/2e58dd7f-ebf0-46f5-9869-79cfe0fcb4a6

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

Just refactoring, nothing players will notice.

### CODE REVIEW

**Comment by tom-leys** [comment]
@0x6273 - I've split the firelock branch #16189 and have made the fixes you suggested there on this branch which is just refactoring.

File: Content.Server/Doors/Systems/AirlockSystem.cs
Comment (AJCM-git) [inline Content.Server/Doors/Systems/AirlockSystem.cs:188]: proxy method

File: Content.Server/Doors/Systems/DoorSystem.cs
Comment (AJCM-git) [inline Content.Server/Doors/Systems/DoorSystem.cs:264]: same

File: Content.Shared/Doors/Components/DoorBoltComponent.cs
Comment (AJCM-git) [inline Content.Shared/Doors/Components/DoorBoltComponent.cs:13]: Access

File: Content.Shared/Doors/Components/DoorBoltComponent.cs
Comment (AJCM-git) [inline Content.Shared/Doors/Components/DoorBoltComponent.cs:25]: VV

File: Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs
Comment (AJCM-git) [inline Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs:6]: you need to have a an empty clientside system inheriting this for prediction to work

File: Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs
Comment (AJCM-git) [inline Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs:12]: i dont think Popup is used on the server so make it private, also newline

**Review by AJCM-git — CHANGES_REQUESTED** [review]
I dont really get whats the point of separating something so connected to airlocks but eh, also, doors dont have wires and dont do much with bolts except handling some logic for airlocks so this has to either be renamed jammed instead of bolted or rename the component/system airlockbolt

**Comment by tom-leys** [comment]
> I dont really get whats the point of separating something so connected to airlocks but eh, also, doors dont have wires and dont do much with bolts except handling some logic for airlocks so this has to either be renamed jammed instead of bolted or rename the component/system airlockbolt

I'm adding bolts to Firelocks https://github.com/space-wizards/space-station-14/pull/16189 - these are not airlocks.

File: Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs
Comment (tom-leys) [reply Content.Shared/Doors/Systems/SharedDoorBoltSystem.cs:12]: Popup from server to client works because I have it in the serverside zombie code.

**Comment by tom-leys** [comment]
I'm now going to wait for a code review before I merge master into this branch. I still want to add firefighting remotes with the ability to block stupid bystanders from releasing fire into public areas.

**Review by AJCM-git — APPROVED** [review]
Haven't tested in-game but the code looks fine, a video showcasing it in the pr body would help (also resolve conflicts and the test error)

**Comment by tom-leys** [comment]
Thanks! :)

**Pull request merged**
